server/lib/whispers/whisper-models.ts

```diff
@@ -0,0 +1,167 @@
+import sql from 'sql-template-strings'
+import { WhisperMessageData } from '../../../common/whispers'
+import db from '../db'
+
+export interface WhisperSessionEntry {
+  targetUserId: number
+  targetUserName: string
+}
+
+export async function getWhisperSessionsForUser(userId: number): Promise<WhisperSessionEntry[]> {
+  const { client, done } = await db()
+  try {
+    const result = await client.query(sql`
+      SELECT u.name AS target_name, u.id AS target_id FROM whisper_sessions
+      INNER JOIN users AS u ON target_user_id = u.id
+      WHERE user_id = ${userId}
+      ORDER BY start_date;
+    `)
+    return result.rows.map(row => ({
+      targetUserId: row.target_id,
+      targetUserName: row.target_name,
+    }))
+  } finally {
+    done()
+  }
+}
+
+export async function startWhisperSession(userId: number, targetUserId: number): Promise<void> {
+  const { client, done } = await db()
+  try {
+    await client.query(sql`
+      WITH ut AS (
+        SELECT u.id AS user_id, t.id AS target_user_id
+        FROM users AS u, users AS t
+        WHERE u.id = ${userId} AND t.id = ${targetUserId}
+      )
+      INSERT INTO whisper_sessions (user_id, target_user_id, start_date)
+      SELECT ut.user_id, ut.target_user_id, CURRENT_TIMESTAMP AT TIME ZONE 'UTC' FROM ut
+      WHERE NOT EXISTS (
+        SELECT 1
+        FROM whisper_sessions
+        WHERE user_id = ut.user_id AND target_user_id = ut.target_user_id
+      );
+    `)
+  } finally {
+    done()
+  }
+}
+
+export async function closeWhisperSession(userId: number, targetUserName: string): Promise<void> {
+  const { client, done } = await db()
+  try {
+    const result = await client.query(sql`
+      WITH tid AS (
+        SELECT t.id AS target_user_id
+        FROM users AS t
+        WHERE t.name = ${targetUserName}
+      )
+      DELETE FROM whisper_sessions
+      WHERE user_id = ${userId} AND target_user_id = (SELECT target_user_id FROM tid);
+    `)
+    if (result.rowCount < 1) {
+      throw new Error('No rows deleted')
+    }
+  } finally {
+    done()
+  }
+}
+
+export interface DbWhisperMessage {
+  msgId: string
```

Comment:
In these structures I would try *very hard* to match what the names are in the database (with camel case instead of snake case, obviously). If you start changing stuff it makes using `Dbify` hard/impossible and you're gonna have a bad time